internal/services/containerapps/container_app_custom_domain_resource.go

```diff
@@ -211,11 +211,22 @@ func (a ContainerAppCustomDomainResource) Read() sdk.ResourceFunc {
 						state.Name = id.CustomDomainName
 						state.ContainerAppId = containerAppId.ID()
 						if pointer.From(v.CertificateId) != "" {
-							certId, err := managedenvironments.ParseCertificateIDInsensitively(pointer.From(v.CertificateId))
-							if err != nil {
-								return err
+							// The `v.CertificateId` returned from API has two possible values. when using an Azure created Managed Certificate,
+							// its format is "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.App/managedEnvironments/%s/managedCertificates/%s",
+							// another format is "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.App/managedEnvironments/%s/certificates/%s",
```

Comment:
Thanks for your feedback. The code has been updated. Could you please take another look?

Reply:
Since [it's also possible to create/delete/read Managed Certificates](https://learn.microsoft.com/en-us/rest/api/containerapps/managed-certificates?view=rest-containerapps-2024-03-01) I suspect we'll need an associated data source/resource to manage those, so that this is usable?

Reply:
I assume that this parsing issue is introduced by [support the ability to use Azure Managed Certificates PR](https://github.com/hashicorp/terraform-provider-azurerm/pull/25356/files). Per the PR's description "This is required to support the automatic creation of Azure Managed Certificates. This PR is in Draft as we are considering design options and may significantly change how this is implemented before inclusion in the provider", I assume that there may be a special reason for not using [Managed Certificates API ](https://learn.microsoft.com/en-us/rest/api/containerapps/managed-certificates?view=rest-containerapps-2024-03-01), although I don't know what the reason is. Now that TF already supports Managed Certificates through `azurerm_container_app_custom_domain`, is it possible to fix the parsing error to unlock the user first?

Reply:
> Now that TF already supports Managed Certificates through azurerm_container_app_custom_domain, is it possible to fix the parsing error to unlock the user first?

Does `azurerm_container_app_custom_domain` really support managed certificates? From what I understand from the [documentation](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/container_app_custom_domain) the support only extends to the possibility to add them manually.

Reply:
I agree with @sinbai that this parsing error should be fixed first to support the user. We are using `azurerm_container_app_custom_domain` to suppress change on the certificate-related fields, so we can manually click through Azure portal to add the managed certificate; until managed certificate is supported in TF.